src/validation/forks.cpp

```diff
@@ -97,6 +96,12 @@ bool IsNov152018Activated(const Consensus::Params &consensusparams, const CBlock
     return IsNov152018Activated(consensusparams, pindexTip->nHeight);
 }
 
+bool AreWeOnBCHChain()
+{
```

Comment:
Rather than overriding the parameters passed to `IsNov152018Activated`, `AreWeOnBCHChain` should take `consensusparams` and `tip` as arguments.

Reply:
I find the name of this function to be misleading. It is checking if you're on BCH *AND* past Nov 2018 fork time. The need to comment "Nov 15th" several places elsewhere is a symptom of this.

How about putting it in namespace `bch` and calling it `IsNov2018Activated`? Usage would be something like:
```
const uint32_t cds_flag = bch::IsNov2018Activated(params, tip) ?
            SCRIPT_ENABLE_CHECKDATASIG :
            0;
```

Reply:
> I find the name of this function to be misleading. It is checking if you're on BCH AND past Nov 2018 fork time.

Indeed. Originally the check for `IsNov2018Activated` wasn't included in the `AreWeOnBCHChain` function. 

On second thought I wonder if it would have been better to keep them separated. Will reconsider tomorrow. 

Having said I find the idea of using `bch::IsNov2018Activated(params, tip)` a good one. 

If I decide to maintain both condition in the same function I will implement your suggested changes.

Reply:
Previously we used whether a fork time was configured to determine the SV or BCH chains.  But this method is going to be very confusing going forward, especially since the next fork date may not be known.  

At this point, we should add a tweak "svEnabled" (default = false) that chooses the fork rather than using these forktime tweaks.  This will be very understandable to users.  If a fork is being prepared on one chain, we then can create one or two fork time variables that define when the fork will happen.

As differences mount, we may need to move to a separate release.  But I don't think that that is needed for this release.